This is a shot in the dark, but we're seeing an odd HWAsan test failure on the Fuchsia clang bots, and AFAICT, this is the only sanitizer-related change in the blamelist. Does it seem possible that this could cause this error?

https://luci-milo.appspot.com/ui/p/fuchsia/builders/toolchain.ci/clang-linux-arm64/b8684518250515764433/overview

```
# RUN: at line 3
env HWASAN_OPTIONS=disable_allocator_tagging=1:random_tags=0:fail_without_syscall_abi=0:symbolize=0 not  /b/s/w/ir/x/w/llvm_build/runtimes/runtimes-aarch64-unknown-linux-gnu-bins/compiler-rt/test/hwasan/AARCH64/TestCases/Output/hwasan_symbolize_stack_overflow.cpp.tmp/hwasan_overflow 16 2>&1 | /b/s/w/ir/x/w/llvm_build/bin/hwasan_symbolize --symbols /b/s/w/ir/x/w/llvm_build/runtimes/runtimes-aarch64-unknown-linux-gnu-bins/compiler-rt/test/hwasan/AARCH64/TestCases/Output/hwasan_symbolize_stack_overflow.cpp.tmp --index | FileCheck /b/s/w/ir/x/w/llvm-llvm-project/compiler-rt/test/hwasan/TestCases/hwasan_symbolize_stack_overflow.cpp --check-prefixes=CHECK,AFTER0
# executed command: env HWASAN_OPTIONS=disable_allocator_tagging=1:random_tags=0:fail_without_syscall_abi=0:symbolize=0 not /b/s/w/ir/x/w/llvm_build/runtimes/runtimes-aarch64-unknown-linux-gnu-bins/compiler-rt/test/hwasan/AARCH64/TestCases/Output/hwasan_symbolize_stack_overflow.cpp.tmp/hwasan_overflow 16
# executed command: /b/s/w/ir/x/w/llvm_build/bin/hwasan_symbolize --symbols /b/s/w/ir/x/w/llvm_build/runtimes/runtimes-aarch64-unknown-linux-gnu-bins/compiler-rt/test/hwasan/AARCH64/TestCases/Output/hwasan_symbolize_stack_overflow.cpp.tmp --index
# .---command stderr------------
# | Could not find symbols for lib/aarch64-linux-gnu/libc.so.6 (Build ID: 45918bc10b33fd96afc550c98de062dccdf44328)
# `-----------------------------
# executed command: FileCheck /b/s/w/ir/x/w/llvm-llvm-project/compiler-rt/test/hwasan/TestCases/hwasan_symbolize_stack_overflow.cpp --check-prefixes=CHECK,AFTER0
# RUN: at line 4

...

# executed command: FileCheck /b/s/w/ir/x/w/llvm-llvm-project/compiler-rt/test/hwasan/TestCases/hwasan_symbolize_stack_overflow.cpp --check-prefixes=CHECK,AFTER1000
# .---command stderr------------
# | /b/s/w/ir/x/w/llvm-llvm-project/compiler-rt/test/hwasan/TestCases/hwasan_symbolize_stack_overflow.cpp:21:12: error: CHECK: expected string not found in input
# |  // CHECK: Potentially referenced stack object:
# |            ^
# | <stdin>:1:1: note: scanning from here
# | ==1736234==ERROR: HWAddressSanitizer: tag-mismatch on address 0xffffe2500218 at pc 0xaaaad678f124
# | ^
# | <stdin>:10:12: note: possible intended match here
# | Address 0xffffe2500218 is located in stack of thread T0
# |            ^
# | 
# | Input file: <stdin>
# | Check file: /b/s/w/ir/x/w/llvm-llvm-project/compiler-rt/test/hwasan/TestCases/hwasan_symbolize_stack_overflow.cpp
# | 
# | -dump-input=help explains the following input dump.
# | 
# | Input was:
# | <<<<<<
# |             1: ==1736234==ERROR: HWAddressSanitizer: tag-mismatch on address 0xffffe2500218 at pc 0xaaaad678f124 
# | check:21'0     X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
# |             2: READ of size 1 at 0xffffe2500218 tags: 01/00 (ptr/mem) in thread T0 
# | check:21'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# |             3:  #0 main in /b/s/w/ir/x/w/llvm-llvm-project/compiler-rt/test/hwasan/TestCases/hwasan_symbolize_stack_overflow.cpp:19:10 
# | check:21'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# |             4:  #1 0xffff8dac2258 (/lib/aarch64-linux-gnu/libc.so.6+0x22258) (BuildId: 45918bc10b33fd96afc550c98de062dccdf44328) 
# | check:21'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# |             5:  #2 0xffff8dac2338 (/lib/aarch64-linux-gnu/libc.so.6+0x22338) (BuildId: 45918bc10b33fd96afc550c98de062dccdf44328) 
# | check:21'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# |             6:  #3 _start in ??:0:0 
# | check:21'0     ~~~~~~~~~~~~~~~~~~~~~
# |             7:  
# | check:21'0     ~
# |             8:  
# | check:21'0     ~
# |             9: Cause: stack tag-mismatch 
# | check:21'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
# |            10: Address 0xffffe2500218 is located in stack of thread T0 
# | check:21'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# | check:21'1                ?                                             possible intended match
# |            11: Thread: T0 0xeffe00002000 stack: [0xffffe1d01000,0xffffe2501000) sz: 8388608 tls: [0xffff8dd644c0,0xffff8dd65360) 
# | check:21'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# |            12:  
# | check:21'0     ~
# |            13:  
# | check:21'0     ~
# |            14: Memory tags around the buggy address (one tag corresponds to 16 bytes): 
# | check:21'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# |            15: Tags for short granules around the buggy address (one tag corresponds to 16 bytes): 
# | check:21'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# |             .
# |             .
# |             .
# | >>>>>>
# `-----------------------------
# error: command failed with exit status: 1

```

--- a/compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp
+++ b/compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp
@@ -307,12 +307,12 @@ int internal_madvise(uptr addr, uptr length, int advice) {
 }
 
 uptr internal_close_range(fd_t lowfd, fd_t highfd, int flags) {
-#    if SANITIZER_FREEBSD
+#    if SANITIZER_FREEBSD || (SANITIZER_LINUX && defined(__NR_close_range))
   return internal_syscall(SYSCALL(close_range), lowfd, highfd, flags);
 #    endif
-  // TODO: Add Linux support using __NR_close_range (available since 5.9).
   return -1;  // Not supported.
 }
+
 uptr internal_close(fd_t fd) { return internal_syscall(SYSCALL(close), fd); }
 
 uptr internal_open(const char *filename, int flags) {

